sbt

```diff
@@ -116,13 +116,20 @@ jar_url () {
 download_url () {
   local url="$1"
   local jar="$2"
+  local exit_code 
   mkdir -p $(dirname "$jar") && {
     if command -v curl > /dev/null; then
-      curl --silent -L "$url" --output "$jar"
+      curl --fail --silent -L "$url" --output "$jar"
+      exit_code=$?
     elif command -v wget > /dev/null; then
       wget --quiet -O "$jar" "$url"
+      exit_code=$?
     fi
-  } && [[ -f "$jar" ]]
```

Comment:
The result of a `{ ... }` block is decided by its final expression - and an `if ... fi` statement always succeeds (in contrast to e.g. Scala). Consequently, that `&&` was always true.